ops/pebble.py

```diff
@@ -2483,8 +2483,8 @@ def __init__(
         self._max_lookahead = max_lookahead
         self._max_boundary_length = max_boundary_length
 
-        self._buf = bytearray()
-        self._pos = 0  # current position in buf
+        self._buf: bytearray = bytearray()
+        self._pos: int = 0  # current position in buf
```

Comment:
Yeah, I couldn't figure this out either. I just spent a bunch of time on it creating a cut-down reproducer ... and it turns out it's a bug in Pyright being confused about Unknown types in some cases. It was a weird one to reproduce because I'd take out a code block or if statement that was seemingly unrelated, and Pyright would suddenly be OK. But it was fixed with Pyright [1.1.290](https://github.com/microsoft/pyright/releases/tag/1.1.290), so I upgraded Pyright to the latest (1.1.291) and removed those type annotations now.

It was likely [this Pyright fix](https://github.com/microsoft/pyright/commit/bd010d11926856025835c3bf75d0351094417e9f) which has to do with caching of Unknown types or something.

Anyway, thanks for nerd-sniping me. ;-)